### ISSUE TITLE
Changing visibility for a node in an editor dock has no effect

### ISSUE THREAD

**Issue by mr11stein** [godotengine/godot#89211]
### Tested versions

- Reproducible in v4.3.dev4.official [df78c0636]
- Not reproducible in v4.2

### System information

Godot v4.3.dev4 - Windows 10.0.19045 - Vulkan (Mobile)

### Issue description

In plugins, if you toggle the visibility of a dock-attached node from false to true, the node stays invisible. Please see the attached project. The label in the dock on the bottom right corner should become visible when the Polygon2D node is selected in the scene tree. This works in 4.2, but not in the latest preview build.

### Steps to reproduce

From an editor plugin, add a node in one of the editor dock slots. Toggle visibility to false and back to true. The node will not be visible, even if the "visible" is set to true.

### Minimal reproduction project (MRP)

[VisibilityBug.zip](https://github.com/godotengine/godot/files/14508736/VisibilityBug.zip)

**Comment by KoBeWi** [comment]
It doesn't make sense .-.
```GDScript
	if is_instance_valid(side_panel_instance):
		side_panel_instance.visible = visible
		print(visible)
		prints(side_panel_instance.visible, side_panel_instance.is_visible_in_tree(), side_panel_instance.get_child(0).visible, side_panel_instance.get_child(0).is_visible_in_tree())
```
This prints:
```
true
true true true false
false
true true true false
```
Which means:
- `side_panel_instance` is visible and visible in tree
- its child is visible, but it's not visible in tree
- when you set `visible` of `side_panel_instance` to false, nothing of the above changes

That said, the `side_panel_instance` is inside TabContainer, which manages children visibility. It might explain why making the content invisible has no effect (and you really shouldn't be doing it with docks), but does not explain why the child would be invisible.

**Comment by AlessandroCampagna** [comment]
Hi I'm new here and would like to work on this.

**Comment by mr11stein** [comment]
> That said, the `side_panel_instance` is inside TabContainer, which manages children visibility. It might explain why making the content invisible has no effect (and you really shouldn't be doing it with docks), but does not explain why the child would be invisible.

What should I do instead? Add/remove the side_panel_container?
Still, it was working in 4.2, so there might be a regression, no?

**Comment by KoBeWi** [comment]
> Still, it was working in 4.2, so there might be a regression, no?

Yeah this is a bug from TabContainer changes. It does have logic to handle this case, but something works incorrectly here.

> Hi I'm new here and would like to work on this.

That's not really a suitable issue for newcomers.

**Comment by AlessandroCampagna** [comment]
> That's not really a suitable issue for newcomers.

i have to make a contribuiton to godot for university and this issue was the most doable one i found
